core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -68,7 +67,12 @@ public Suggest() {
     }
 
     public Suggest(List<Suggestion<? extends Entry<? extends Option>>> suggestions) {
+        // we sort suggestions by their names to ensure iteration over suggestions are consistent
+        // this is needed as we need to fill in suggestion docs in SearchPhaseController#sortDocs
+        // in the same order as we enrich the suggestions with fetch results in SearchPhaseController#merge
+        suggestions.sort((o1, o2) -> o1.getName().compareTo(o2.getName()));
```

Comment:
I am confused why do we need to sort? Is this due to suggestions from different shards can come back in different order?